src/components/AppWrapper.tsx

```diff
@@ -531,6 +637,44 @@ const AppWrapper: React.FC<Props> = (props: Props): React.ReactElement => {
     );
   }
 
+  /* ------------- Prompt ------------ */
+
+  if (promptInfo) {
+    modal = (
+      <ModalForWrapper
+        key={`prompt-${promptInfo.title}-${promptInfo.text}`}
+        title={promptInfo.title}
+        type={ModalType.OkayCancel}
+        okayLabel={promptInfo.opts.confirmButtonText}
+        okayVariant={promptInfo.opts.confirmButtonVariant}
+        cancelLabel={promptInfo.opts.cancelButtonText}
+        cancelVariant={promptInfo.opts.cancelButtonVariant}
+        onClose={(buttonType) => {
+          const result = buttonType === ModalButtonType.Okay ? currentPromptInputFieldText : null;
+          setPromptInfo(undefined);
+          if (onPromptClosed) {
+            onPromptClosed(result);
+          }
+        }}
+        onTopOfOtherModals
+        dontAllowBackdropExit
+      >
+        <div className="d-flex flex-column align-items-center">
+          <p>{promptInfo.text}</p>
+          <input
+            type="text"
+            placeholder={promptInfo.opts.placeholder}
+            value={currentPromptInputFieldText}
+            onChange={(e) => {
+              return setCurrentPromptInputFieldText(e.target.value);
+            }}
+          />
+        </div>
+
+      </ModalForWrapper>
+    );
+  }
+
```

Comment:
Sorry, just thought about this now, but could you also include logic for input field validation?

Two additional prompt info opts:

1. minNumChars: number – if the text field has fewer chars than this number, underneath the text field, have red text that explains that the text needs to be at least X chars

2. findValiationError: (text: string) => string | undefined – if the functions is defined, call the function on each render. If the function returns a string (this is the explanation) then display that explanation under the text box in that red text area. Example explanation: "You must include at least one uppercase letter". If the validator function doesn't return anything, then don't show any error

General notes:
- If both props are included, then perform _both_ types of validation
- When there's a validation error, remove the okay button (the only option should be to cancel)